src/librustdoc/json/conversions.rs

```diff
@@ -589,6 +589,8 @@ impl FromWithTcx<clean::Type> for Type {
                 traits: bounds.into_tcx(tcx),
             }),
             Generic(s) => Type::Generic(s.to_string()),
+            // FIXME: add dedicated variant to json Type?
+            SelfTy => Type::Generic("Self".to_owned()),
```

Comment:
cc @aDotInTheVoid -- what would you like to do here?

Reply:
This is fine for now as it preserves the existing behavior, but yeah this should almost certainly be changed in a future PR

Reply:
after some discussion with @compiler-errors, i’m not sure the current json behaviour is wrong. but either way, this is fine for this PR

Reply:
Yes, technically speaking `Self` is either an alias (in the case of concrete impls) or a type parameter (in the case of traits). So what we're doing here in Rustdoc is more of a convenience for how users perceive `Self`, rather than how the compiler does.